WordPress/wordpress-playground
# Onboarding modal

## What?

* Trying to make client onboarding experience better with this [#497](https://github.com/WordPress/wordpress-playground/issues/497) ticket.
* This code changes will provide a  friendly message(modal experience) to clients that the changes they are making are private and staying within this session. If you refresh your page, all your changes will go away.
* I have introduced new flag called `isOnboarded` inside browser local storage. If this flag is not available into user's browser local storage then it will open modal and if it is present then it will not.

## Why?


* Better client experience and providing a friendly reminder to clients regarding they activities are temporary so they do not hit refresh by mistake and lose their session data.

## How?


* Create custom `React Modal Component` which accepts React Children component with custom message. This modal component has close button inside too so client can easily close the modal after read. 
* Create `util` folder and added utility to check whether the client is onboarded for the first time or not.

## Testing Instructions





1. Please checkout to my branch `playground/Onboard_enhacement`
2. Install all the dependencies by `npm install`
3. Please run `npm run dev`
4. You will see the modal experienced. This modal is responsive too.
5. Once you close the modal, it will not prompt again if you refresh your browser. 
6. If you still would like to see the modal for testing purposes, please open `dev-tools` and go to `Application` tab and delete the flag "isOnboarded" from `local storage` and then refresh the page. You will be able to see the Modal component again.

## Client Experience 


https://github.com/WordPress/wordpress-playground/assets/5431237/325febb4-7582-4260-a35a-df69ecad9dd8
